tooling/miniutil/src/build/ty.rs

```diff
@@ -70,10 +70,11 @@ pub fn enum_variant(ty: Type, tagger: &[(Offset, u8)]) -> Variant {
     }
 }
 
-pub fn enum_ty(variants: &[Variant], discriminator: Discriminator, size: Size, align: Align) -> Type {
+pub fn enum_ty(variants: &[Variant], discriminator: Discriminator, discriminant_bytes: impl Into<Int>, size: Size, align: Align) -> Type {
```

Comment:
Wouldn't it be better to give the discriminant type as a type? Callers would then do something like `<u8>::get_type()`. Or maybe make `enum_ty` take a `T: TypeConv` then one would do `enum_ty::<u8>(...)`.

Reply:
I like the idea of making `enum_ty` take a `DiscriminantTy: TypeConv` as this type is part of the enum. (I also realized that the current way doesn't allow signed integer discriminants without the bare-bones construction.)